crates/rpc-types-engine/src/forkchoice.rs

```diff
@@ -34,19 +34,20 @@ pub struct ForkchoiceState {
 ///
 /// These are considered hard RPC errors and are _not_ returned as [PayloadStatus] or
 /// [PayloadStatusEnum::Invalid].
-#[derive(Clone, Copy, Debug, PartialEq, Eq, thiserror::Error)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
 pub enum ForkchoiceUpdateError {
```

Comment:
thiserror also takes care of Display,

I think it would be better to manually impl Display and then just feature gate the error impl